flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/model/KinesisStreamShard.java

```diff
@@ -61,19 +68,23 @@ public String getShardId() {
 	}
 
 	public String getStartingSequenceNumber() {
-		return shard.getSequenceNumberRange().getStartingSequenceNumber();
+		SequenceNumberRange sequenceNumberRange = shard.getSequenceNumberRange();
+		return (sequenceNumberRange == null) ? null : sequenceNumberRange.getStartingSequenceNumber();
```

Comment:
If this is the case, than I don't see the reason for these checks.
In general, I have the feeling that the `KinesisStreamShard` is overly complicated. Maybe its easier to just have a method getShard() instead of proxying everything?